analytics/analytics/server.py

```diff
@@ -0,0 +1,207 @@
+import logging
+import sys
+from uuid import UUID
+
+import falcon
+from falcon_cors import CORS
+from spectree import Response, SpecTree
+
+from analytics import settings
+from analytics.docs.schemas.requests import (
+    DetailEventSchema,
+    ResultClickEventSchema,
+    SearchEventSchema,
+    SearchRatingEventSchema,
+)
+from analytics.docs.schemas.responses import (
+    BadRequestSchema,
+    CreatedSchema,
+    InternalServerErrorSchema,
+    OkSchema,
+)
+from analytics.event_controller import EventController
+
+
+with open("docs/redoc.html", "r") as redoc_html:
+    page_templates = {"redoc": redoc_html.read()}
+with open("docs/README.md", "r") as readme_file:
+    description = readme_file.read()
+spec = SpecTree(
+    "falcon",
+    title="Openverse Analytics",
+    version=settings.VERSION,
+    description=description,
+    page_templates=page_templates,
+    path="doc",
+    mode="strict",
+    validation_error_status=400,
+)
+
+resp = Response(
+    HTTP_201=CreatedSchema,
+    HTTP_400=BadRequestSchema,
+    HTTP_500=InternalServerErrorSchema,
+)
+
+
+class HealthResource:
+    @spec.validate(
+        resp=Response(HTTP_200=OkSchema),
+        validation_error_status=200,  # never fails validation
+        tags=["health"],
+    )
+    def on_get(self, _, resp):
+        """detail"""
+        resp.media = {"status": "200 OK"}
+        resp.status = falcon.HTTP_200
+
+
+class BaseEventResource:
+    """Base class for all resource that need access to an event controller"""
+
+    @staticmethod
+    def _validate_uuid(field, uuid):
+        try:
+            return UUID(uuid, version=4).hex
+        except ValueError:
+            raise falcon.HTTPBadRequest(description=f"{field} must be a v4 UUID")
+
+    def __init__(self, event_controller: EventController, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.event_controller = event_controller
+
+
+class SearchEventResource(BaseEventResource):
+    @spec.validate(
+        json=SearchEventSchema,
+        resp=resp,
+        tags=["search_event"],
+    )
+    def on_post(self, req, resp):
+        """create
```

Comment:
A lot of these docstrings start with `"""create`, is it alright to remove that bit?

Reply:
These are used to give operation IDs to the endpoints (shows up in the ReDoc as headings, which would otherwise default to `on_post <POST>`)

Reply:
Got it! That makes sense, I just wasn't aware :slightly_smiling_face: My naive assumption was copypasta leftovers :grin: